src/main/test/surfrdUtils_test/test_surfrdUtils.pf

```diff
@@ -43,6 +43,114 @@ contains
+     ! updated wt_lunit values.
+     ! Tests performed:
+     ! - toosmall_* = 0 ... DO NOTHING
+     ! - toosmall_* = 1 ... Remove landunits with <= 1% of the gridcell area
+     ! - toosmall_* = 8 ... Remove landunits with <= 8% of the gridcell area
+     ! - toosmall_* = 39 ... Remove landunits with <= 39% of the gridcell area
+     ! - toosmall_* = 100 ... Keep only the dominant landunit per gridcell
+     use pftconMod, only: pftcon
+     use clm_instur, only: wt_lunit
+     use landunit_varcon, only: max_lunit, istsoil, istcrop, istice_mec, &
+                                istdlak, istwet, isturb_tbd, isturb_hd, &
+                                isturb_md
+
+     implicit none
+     integer :: test  ! test number in loop of tests
+     integer, parameter :: begg = 2, endg = 4, tests = 5
+     integer, parameter :: toosmall_soil(tests) = (/ 0, 1, 8, 39, 100/)
+     integer, parameter :: toosmall_crop(tests) = (/ 0, 1, 8, 39, 100/)
+     integer, parameter :: toosmall_glacier(tests) = (/ 0, 1, 8, 39, 100/)
+     integer, parameter :: toosmall_lake(tests) = (/ 0, 1, 8, 39, 100/)
+     integer, parameter :: toosmall_wetland(tests) = (/ 0, 1, 8, 39, 100/)
+     integer, parameter :: toosmall_urb_tbd(tests) = (/ 0, 1, 8, 39, 100/)
+     integer, parameter :: toosmall_urb_hd(tests) = (/ 0, 1, 8, 39, 100/)
+     integer, parameter :: toosmall_urb_md(tests) = (/ 0, 1, 8, 39, 100/)
+     real(r8), allocatable :: wt_expected(:,:)
+     real(r8), allocatable :: wt_in_out(:,:)  ! used in subr. call
+     real(r8) :: expctd(9)
+     real(r8), parameter :: input_value(8) = (/40._r8, 33._r8, 12._r8, 7._r8, 5._r8, 2._r8, 1._r8, 0._r8/)
+
+     ! Set relevant pftcon values to defaults; override where necessary
+     call pftcon%InitForTesting()
+     allocate( wt_lunit(begg:endg,istsoil:max_lunit) )
+     allocate( wt_expected(begg:endg,istsoil:max_lunit) )
+     allocate( wt_in_out(begg:endg,istsoil:max_lunit) )
+
+     ! INPUT VALUES
+     wt_lunit(begg:,:) = 0._r8  ! initialize
+     wt_lunit(begg:,istsoil) = (/ input_value(1), input_value(8), input_value(4)/)
+     wt_lunit(begg:,istcrop) = (/ input_value(2), input_value(7), input_value(5)/)
+     wt_lunit(begg:,istice_mec) = (/ input_value(3), input_value(6), input_value(6)/)
+     wt_lunit(begg:,istdlak) = (/ input_value(4), input_value(5), input_value(7)/)
+     wt_lunit(begg:,istwet) = (/ input_value(5), input_value(4),  input_value(8)/)
+     wt_lunit(begg:,isturb_tbd) = (/ input_value(6),  input_value(3), input_value(1)/)
+     wt_lunit(begg:,isturb_hd) = (/ input_value(7),  input_value(2), input_value(2)/)
+     wt_lunit(begg:,isturb_md) = (/ input_value(8),  input_value(1), input_value(3)/)
+     wt_lunit(:,:) = wt_lunit(:,:) / 100._r8
+     call check_sums_equal_1( wt_lunit, begg, "test_check_sums_add_to_1", &
+                              "should not trigger an error")
+
+     do test = 1, tests
+
+        ! OUTPUT VALUES EXPECTED
+        if (test == 1) then
```

Comment:
Little more documentation on what the different tests are would be good here (and possibly making the hardcoded test numbers into parameters with a meaning could be helpful).

Reply:
Done.